site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace app\controllers;
+
+use app\libraries\Core;
+use app\libraries\response\Response;
+use app\libraries\response\WebResponse;
+use app\libraries\response\JsonResponse;
+use app\libraries\response\RedirectResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use app\models\OfficeHoursQueueStudent;
+use app\libraries\routers\AccessControl;
+
+/**
+ * Class OfficeHoursQueueController
+ *
+ */
+class OfficeHoursQueueController extends AbstractController {
+
+    public function __construct(Core $core) {
+        parent::__construct($core);
+    }
+
+    /**
+     * @param
+     * @Route("/{_semester}/{_course}/OfficeHoursQueue")
+     * @return Response
+     */
+     public function showQueue(){
+       if(!$this->core->getConfig()->isQueueEnabled()){
+         return Response::RedirectOnlyResponse(
+             new RedirectResponse($this->core->buildCourseUrl(['home']))
+         );
+       }
+       if(!$this->core->getUser()->accessGrading()){
+         $oh_queue = $this->core->getQueries()->getQueueByUser($this->core->getUser()->getId());
+         return Response::WebOnlyResponse(
+           new WebResponse(
+             'OfficeHoursQueue',
+             'showQueueStudent',
+             $oh_queue,
+             $this->core->getConfig()->getCourse()
+             )
+           );
+        }
+        else{
+          $oh_queue = $this->core->getQueries()->getInstructorQueue();
+          return Response::WebOnlyResponse(
+            new WebResponse(
+              'OfficeHoursQueue',
+              'showQueueInstructor',
+              $oh_queue,
+              $this->core->getConfig()->getCourse()
+              )
+            );
+        }
+     }
+     /**
+      * @param
+      * @Route("/{_semester}/{_course}/OfficeHoursQueue/add", methods={"POST"})
+      * @return Response
+      */
+      public function addPerson(){
+        $section_id = $this->core->getQueries()->isValidCode($_POST['code']);
```

Comment:
Please perform a verification that the keys actually exists in the `$_POST` array and do a step of:
```
if (!isset($_POST['code']) || !isset($_POST['name'])) {
    $this->core->addErrorMessage('Missing name or code in request');
}
```